src/Network/HTTP/Affjax.purs

```diff
@@ -0,0 +1,158 @@
+module Network.HTTP.Affjax
+  ( Ajax()
+  , Affjax()
+  , AffjaxOptions()
+  , AffjaxResponse()
+  , url, method, content, headers, username, password
+  , affjax
+  , affjax'
+  , get
+  , post, post_
+  , put, put_
+  , delete, delete_
+  ) where
+
+import Control.Monad.Aff (Aff(), makeAff)
+import Control.Monad.Eff (Eff())
+import Control.Monad.Eff.Exception (Error(), error)
+import Data.Either (Either(..))
+import Data.Foreign (Foreign(..), F())
+import Data.Function (Fn4(), runFn4)
+import Data.Options (Option(), Options(), options, (:=), opt)
+import Network.HTTP.Affjax.Request
+import Network.HTTP.Affjax.Response
+import Network.HTTP.Affjax.ResponseType
+import Network.HTTP.Method (Method(..))
+import Network.HTTP.RequestHeader (RequestHeader())
+import Network.HTTP.ResponseHeader (ResponseHeader(), responseHeader)
+import Network.HTTP.StatusCode (StatusCode())
+
+-- | The effect type for AJAX requests made with Affjax.
+foreign import data Ajax :: !
+
+-- | The type for Affjax requests.
+type Affjax e a = Aff (ajax :: Ajax | e) (AffjaxResponse a)
+
+-- | Options type for Affjax requests.
+foreign import data AffjaxOptions :: *
+
+-- | The type of records that will be received as an Affjax response.
+type AffjaxResponse a =
+  { status :: StatusCode
+  , headers :: [ResponseHeader]
+  , response :: a
+  }
+
+-- | Sets the URL for a request.
+url :: Option AffjaxOptions String
+url = opt "url"
+
+-- | Sets the HTTP method for a request.
+method :: Option AffjaxOptions Method
+method = opt "method"
+
+-- | Sets the content to send in a request.
+content :: Option AffjaxOptions RequestContent
+content = opt "content"
+
+-- | Sets the headers to send with a request.
+headers :: Option AffjaxOptions [RequestHeader]
+headers = opt "headers"
+
+-- | Sets the HTTP auth username to send with a request.
+username :: Option AffjaxOptions String
+username = opt "username"
+
+-- | Sets the HTTP auth password to send with a request.
+password :: Option AffjaxOptions String
+password = opt "password"
+
+-- | Sets the expected response type for a request. This is not exposed outside
+-- | of the module as the `ResponseType` is set based on the `Responsable`
+-- | instance for the expected result content type.
+responseType = opt "responseType" :: Option AffjaxOptions ResponseType
+
+-- | Runs a request.
+affjax :: forall e a. Responsable a ->
```

Comment:
FYI v0.8.0 of Aff will allow asynchronous computations to be canceled if supported by the underlying implementation, which we can take advantage of in AJAX requests to unnecessary requests in some cases.